src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -941,7 +959,7 @@ UCS_CLASS_INIT_FUNC(uct_rdmacm_cm_t, uct_component_h component,
     status = ucs_async_set_event_handler(worker_priv->async->mode,
                                          self->ev_ch->fd, UCS_EVENT_SET_EVREAD,
                                          uct_rdmacm_cm_event_handler, self,
-                                         worker_priv->async);
+                                         NULL);
```

Comment:
nope, events will be handled in next event_handler invocation

Reply:
but then it will work great if progress thread do nothing related to UCX (or yields), but if it calls empty `ucp_worker_progress` in busy waiting (some recv, for example), then the lock might never (or too rare) be acquired by async thread.
maybe add some counter for try_block and then fallback to block?